pip_audit/_service/interface.py

```diff
@@ -11,10 +11,11 @@
 from packaging.version import Version
 
 
-@dataclass(frozen=True)
-class Dependency:
+# MyPy doesn't like abstract dataclasses so silence the checker here
+@dataclass(frozen=True)  # type: ignore[misc]
+class Dependency(ABC):
```

Comment:
Let's de-abstract this and turn `is_skipped` into an instance check, so that MyPy can follow through the types here (and hopefully, if it's smart enough, figure out our subclass dispatch on `is_skipped` elsewhere).

I'll commit the change for that.

Reply:
Looks like MyPy isn't smart enough to follow through the test here, unfortunately, probably because Python doesn't have a really sound way of enforcing strict subclassing. So I've added some runtime checks as unit tests instead.